### PR TITLE
gui: fix mandatory/leisure detection of upgrade check 

### LINKED ISSUE #1949: Pop up notice about available update says update is leisure when its mandatory.
Got this today, should be fixed to avoid confusion. People wouldn't know otherwise unless they clicked the more details button.

https://gyazo.com/7c20e5971bc5f4ea6396ffa06aa4b1a5

### PR DESCRIPTION
**Pull request by Pythonix**
This should close https://github.com/gridcoin-community/Gridcoin-Research/issues/1949

### CODE REVIEW

File: src/gridcoin/upgrade.cpp [hunk 1]
```diff
@@ -95,10 +95,10 @@ bool Upgrade::CheckForLatestUpdate(bool ui_dialog, std::string client_message_ou
         return false;
     }
 
-    if (GithubReleaseTypeData.find("leisure"))
+    if (GithubReleaseTypeData.find("leisure") != std::string::npos)
```
Comment (cyrossignol) [inline src/gridcoin/upgrade.cpp:100]: This check also suffers from a case-sensitive comparison. One possible solution:

```suggestion
    boost::to_lower(GithubReleaseTypeData);
    
    if (GithubReleaseTypeData.find("leisure") != std::string::npos)
```

...which will need:
 
```c++
#include <boost/algorithm/string/case_conv.hpp>
```

Comment (Pythonix) [reply src/gridcoin/upgrade.cpp:100]: Done =)

**Review by cyrossignol — APPROVED** [review]
Looks good.

**Review by jamescowens — APPROVED** [review]
npos problem fixed. Case insensitive now. ACK.

**Pull request merged**